apps/meteor/client/views/hooks/useMemberExists.ts

```diff
@@ -0,0 +1,12 @@
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+
+type UseMemberExistsProps = { roomId: string; username: string; isMember?: boolean };
+
+export const useMemberExists = ({ roomId, username, isMember }: UseMemberExistsProps) => {
```

Comment:
While this would work, but imo this doesn't follow best design principles and best practices.

We are sending the info whether a user is part of a room or not to a hook which is supposed to give us this info.

Why don't we move this hook call to component above in the tree, like `UserCardWithData` where it would return us this info which can be used when calling the `userInfoAction` hook. 🤔